File: clients/javascript/src/models/msgStreamReceiveIn.ts
// this file is @generated

export interface MsgStreamReceiveIn {
    namespace?: string | null;
    batchSize?: number;
    leaseDurationMillis?: number;
    defaultStartingPosition?: string;
}

export interface MsgStreamReceiveIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    batchSize?: number;
    leaseDurationMillis?: number;
    defaultStartingPosition?: string;
}

export const MsgStreamReceiveInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgStreamReceiveIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            batchSize: object['batch_size'],
            leaseDurationMillis: object['lease_duration_millis'],
            defaultStartingPosition: object['default_starting_position'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgStreamReceiveIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'batch_size': self.batchSize,
            'lease_duration_millis': self.leaseDurationMillis,
            'default_starting_position': self.defaultStartingPosition,
        };
    }
}

File: clients/rust/src/models/msg_stream_receive_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgStreamReceiveIn {
    pub topic: String,

    pub consumer_group: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub batch_size: Option<u16>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub lease_duration_millis: Option<u64>,
}

impl MsgStreamReceiveIn {
    pub fn new(topic: String, consumer_group: String) -> Self {
        Self {
            topic,
            consumer_group,
            batch_size: None,
            lease_duration_millis: None,
        }
    }
}
